### PR TITLE
fix error TCP Client due to data buffer allocation failed

### PR DESCRIPTION
**Pull request by Desvauxm-st**
subsys: ethernet: fix TCP Client Issue

fix error TCP Client due to data buffer allocation failed
fix bug at high speed
see open issue: Ethernet TCP Client Issue description with iperf/zperf #53885

in case of data buffer allocation failed: 
wait and stay in the loop: a new alloc_buffer is requested again

### CODE REVIEW

File: subsys/net/ip/net_pkt.c [hunk 1]
```diff
@@ -1147,69 +1147,82 @@ int net_pkt_alloc_buffer(struct net_pkt *pkt,
 			 k_timeout_t timeout)
 #endif
 {
-	uint64_t end = sys_clock_timeout_end_calc(timeout);
-	struct net_buf_pool *pool = NULL;
-	size_t alloc_len = 0;
-	size_t hdr_len = 0;
+	uint8_t Max_try_pkt_alloc_buffer = 3;
```
Comment (jukkar) [inline subsys/net/ip/net_pkt.c:1150]: For consistency, please use small letters in vars, so `max_try_...` here

**Review by jukkar — COMMENTED** [review]
Difficult to read the diff because of the whitespace warning so did only superficial review. Anyway, I am just wondering about this fix, would it be simpler to just increase the number of tx/rx buffers?

**Comment by Desvauxm-st** [comment]
> Difficult to read the diff because of the whitespace warning so did only superficial review. Anyway, I am just wondering about this fix, would it be simpler to just increase the number of tx/rx buffers?

I only try to change CONFIG_NET_TCP_TIME_WAIT_DELAY and CONFIG_NET_TCP_ACK_TIMEOUT without success
I want to know if the way  suggested is to  increase CONFIG_NET_PKT_RX_COUNT and CONFIG_NET_PKT_TX_COUNT ?

**Review by rlubos — CHANGES_REQUESTED** [review]
I don't think this patch could be accepted. Introducing retry counter makes little sense to me - it introduces confusion wrt `timeout` parameter (i. e. it's no longer reliable) plus, the same effect could be acomplished by providing larger timeout value to the function (if really needed).

I think that regarding #53885 we should rather focus on figuring why does the buffer allocation fail in the first place (could be simply buffer shortage as @jukkar  suggests).

**Comment by Desvauxm-st** [comment]
I change the value of CONFIG_NET_PKT_RX_COUNT, CONFIG_NET_PKT_TX_COUNT, CONFIG_NET_BUF_RX_COUNT
CONFIG_NET_BUF_TX_COUNT and when I increase all the values it works better

**Pull request closed without merging**